x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/state.ts

```diff
@@ -158,6 +160,39 @@ export const getInitialState = (): State => ({
   estimatedModelMemoryLimit: '',
 });
 
+const getExcludesFields = (excluded: string[]) => {
+  const { fields } = newJobCapsService;
+  const updatedExcluded = [];
+
+  for (let i = 0; i < excluded.length; i++) {
+    let isBothTypes = false;
+    const fieldName = excluded[i];
+    updatedExcluded.push(fieldName);
+    const fieldType = fields.find(field => field.name === fieldName)?.type;
+    // If it's a keyword type - check if it has a corresponding text type
+    if (fieldType !== undefined && fieldType === ES_FIELD_TYPES.KEYWORD) {
+      const textTypeName = fieldName.replace(/\.keyword$/, '');
+      const field = newJobCapsService.getFieldById(textTypeName);
+      isBothTypes = field !== null && field.type === ES_FIELD_TYPES.TEXT;
+
+      if (isBothTypes) {
+        updatedExcluded.push(textTypeName);
+      }
+    } else if (fieldType !== undefined && fieldType === ES_FIELD_TYPES.TEXT) {
+      //   If text, check if has corresponding keyword type
+      const keywordTypeName = `${fieldName}.keyword`;
+      const field = newJobCapsService.getFieldById(keywordTypeName);
+      isBothTypes = field !== null && field.type === ES_FIELD_TYPES.KEYWORD;
+
+      if (isBothTypes) {
+        updatedExcluded.push(keywordTypeName);
+      }
+    }
+  }
```

Comment:
I think this should be more generic.  It doesn't have to be the case that the main field is `text` and the `keyword` multi-field has a `.keyword` suffix.  Any suffix can be used, for example `city` and `city.raw` in [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html).

Also, the multi-fields don't have to be `text` and `keyword`.  They could be `integer` and `keyword` or `text` and `text` but with different analyzers.

So I think the logic should be:

1. If a field is excluded that doesn't have a dot in its name then exclude that field plus all others whose names begin with that followed by a dot.  For example, if `foo` is excluded then exclude `foo` and any field whose name matches the regex `foo\..+`.
2. If a field is excluded that has a dot in its name then check if there is a field whose name equals the name with the last dot suffix removed.  If there is then treat that as the excluded field and do the step 1 logic.  Otherwise just exclude the exact field.  For example, if `foo.raw` is excluded and a field `foo` exists then exclude `foo` and any field whose name matches the regex `foo\..+`.  If `foo.raw` is excluded and there is no field called `foo` then just exclude `foo.raw`.

Don't worry about types, just names.

Reply:
Just to update here after a quick chat. This was originally meant to target the most common case (text/keyword field types). I misunderstood the legality of using `.` in field names. I'll be updating this with the above suggestion. Thanks @droberts195 😄

Reply:
Updated in 33d18b55b1e05afcb9946198aa9c6bc7f8a2bac0
cc @droberts195

Reply:
This has been updated with suggested changes cc @droberts195